### PR TITLE
Omnibar

### LINKED ISSUE #5551: Creating an Omnibar
From any where in the app, I want to be able to hit Cmd + K and jump to another chart or dashboard based upon the name. We don't have to use blueprint, open to other libraries. Mainly want to shorthand on moving around the app.

Example:
omnibar

### PR DESCRIPTION
**Pull request by hughhhh**
Inspired by my favorite command in Slack, I added an omnibar to dashboard page to allow users to hop between dashboards in superset. Pressing Cmd + K will trigger the omnibar to appear. After typing, pressing `Enter` will redirect the user to the highlighted dashboard.

omnibar_demo

### Todo:
- [X] Auto place scroll for typing (after hotkey trigger you should be able to type in the bar
- [X] Pressing enter should change the url to that dashboard

### Next:
- Add other resources (charts, queries, tags, etc.)
- Potentially put this as a global hotkey to allow for better exploration across superset

Closes #5551 

@betodealmeida @xtinec @mistercrunch @williaster @graceguo-supercat  @kristw

### CODE REVIEW

**Comment by hughhhh** [comment]
omnibar_demo

File: superset/assets/src/dashboard/components/Dashboard.jsx
Comment (kristw) [inline superset/assets/src/dashboard/components/Dashboard.jsx]: Please do not add new jquery usage.

Comment (williaster) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: yeah this should be a blocker

Comment (williaster) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: You prob need a div ref and https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/focus

File: superset/assets/src/dashboard/components/Dashboard.jsx
Comment (williaster) [inline superset/assets/src/dashboard/components/Dashboard.jsx]: I think this needs to be more robust, what if more modals are added? what happens if the selectors are empty and this throws an error? 

ideally you would get a DOM ref via React`s `ref` api.

File: superset/assets/src/dashboard/components/Dashboard.jsx
Comment (williaster) [inline superset/assets/src/dashboard/components/Dashboard.jsx]: I think you could create a component with both the modal + `OmniContainer` to simplify this component. and also use a react fragment instead of adding an unneeded `div`

```
<React.Fragment>
  <OmniWithModal />
  <DashboardBuilder />
</React.Fragment>
```

**Review by williaster — CHANGES_REQUESTED** [review]
looking good overall, had a couple suggestions to clean it up a little bit more.

File: superset/assets/src/dashboard/components/Dashboard.jsx
Comment (hughhhh) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: What is `React.Fragment` doing for us here?

Comment (xtinec) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: It is a trick to get ride of holder `div`s. 🙂 
See https://reactjs.org/docs/fragments.html

File: superset/assets/src/components/OmniContainer.jsx
Comment (xtinec) [inline superset/assets/src/components/OmniContainer.jsx]: I feel we should move network logic into a separate say `/api/network.ts` file and make proper async actions for triggering them instead of stuffing things in the containers. I've found `redux-saga` to be a wonderful library for managing async actions (i.e. having `saga`s for async actions and `reducer`s for plain action). 

@kristw @williaster what do you guys think?

Btw, if we like the idea. I'm happy to refactor one of the existing containers to get things set up.

File: superset/assets/src/dashboard/components/Dashboard.jsx
Comment (xtinec) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: I agree. The `getElementsByClassName` and such approach tends to be brittle especially when the layout and classnames of the library we use for modal dialogs changes on a future upgrade.

File: superset/assets/src/dashboard/components/Dashboard.jsx
Comment (xtinec) [inline superset/assets/src/dashboard/components/Dashboard.jsx]: I'm not convinced of the value of logging all the keydowns. Each logging call will trigger an rpc to the server. 
https://github.com/apache/incubator-superset/blob/11a7ad00b7c44785752bf23ef90bcf4ebb58598e/superset/assets/src/logger.js#L59

Given how often people may type in a dashboard especially in the presence of filters, I don't think we should log these keydowns and potentially flood the server here.

**Review by xtinec — CHANGES_REQUESTED** [review]
Looking good. Agree with existing suggestions.  A couple additional things to tweak.

File: superset/assets/src/dashboard/components/Dashboard.jsx
Comment (williaster) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: isn't this only sent for the toggling of the bar tho? since it's inside the `control + k` blocks?

Comment (hughhhh) [reply superset/assets/src/dashboard/components/Dashboard.jsx]: I want to verify that this feature is being used before continuing main reason i wanted to add the logger here

**Review by williaster — CHANGES_REQUESTED** [review]
Code is looking okay, this is a huge design change so @kristw and I want to hold off until it's approved by one or all of @elibrumbaugh @thinmatt @dorq

**Comment by williaster** [comment]
I think minimally this should be under a feature flag as well to turn off/on

**Comment by dorq** [comment]
A couple questions to fully understand better:
- Omnibar can _only_ be activated when viewing an existing dashboard?
- Omnibar only searches dashboard names? (Rather than including, say, dashboard chart names, or chart names outside of any dashboard)

**Comment by hughhhh** [comment]
@williaster added the feature flag

**Comment by hughhhh** [comment]
Omnibar can only be activated when viewing an existing dashboard?
```
Yes
```
Omnibar only searches dashboard names? (Rather than including, say, dashboard chart names, or chart names outside of any dashboard)
```
Yes only dashboards. Would love to do another resources (charts, tags, queries, etc.), but want to validate this feature before i get into that.
```

**Comment by elibrumbaugh** [comment]
Hey @hughhhh this is a really neat idea!

A few minor nits:

1. Could we position the search field further down from the top of the screen. Ideally I think it would be centered in the current viewport.
2. I don't think you need the .. after `Search for dashboards` placeholder text.
3. Rephrasing that search prompt to `Search all dashboards` might be a little more clear.

How does this handle unpublished dashboards if that PR goes in?

**Review by williaster — APPROVED** [review]
Thanks Hugh! Lgtm 💯

**Pull request merged**
